### PR TITLE
DDP-5539 Replace `whitelist` with `allowlist`

### PR DESCRIPTION
**Pull request by zyme**
## Context

DDP-5539 is about removing the offensive term `whitelist` and replacing it with `allowlist`.  Although the term still exists in imported library code, this PR replaces all instances of `whitelist` with `allowlist`.

The `dev` secrets have been updated to use the new `allowlist` keys.  Ideally we could first _add_ the new stanza for `ipAllowList` in vault to allow for easier migration, and then once a prod deploy is complete, we could remove the old stanza from vault.

If others find other occurrences of `whitelist`, please let me know and I'll fix them up as well.

## Checklist

- [x ] I have labeled the type of changes involved using the `C-*` labels.
- [ x] I have assessed potential risks and labeled using the `R-*` labels.
- [ x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ x] I have considered security and privacy, and added `I-*` labels as needed
- [ x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ x] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ x] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [ x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/scripts/admin/whitelist-register-route-auth0.sh [hunk 1]
```diff
@@ -1,20 +0,0 @@
-#!/bin/bash
```
Comment (zyme) [inline pepper-apis/scripts/admin/whitelist-register-route-auth0.sh:1]: This file isn't needed--a bit of an experiment with cloud armor from long ago.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/constants/ConfigFile.java [hunk 2]
```diff
@@ -193,4 +193,10 @@
     public static class Auth0LogEvents {
         public static final String AUTH0_LOG_EVENTS_TOKEN = "auth0LogEvents.token";
     }
+
+    public static class Auth0LogEventApi {
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/constants/ConfigFile.java:216]: Looks like an unrelated artifact from another branch. I recommend doing a rebase to `develop`!

**Review by yufengwng — APPROVED** [review]
LGTM

**Comment by zyme** [comment]
@yufengwng I think I've gotten things cleaned up and in synch with `develop`...

File: pepper-apis/src/main/java/org/broadinstitute/ddp/constants/ConfigFile.java (hunk 2 above)
Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/constants/ConfigFile.java:216]: It doesn't look like latest `develop` branch has these changes, but I think it's fine to merge -- it's only constants. However, if you want, you can play with git:

I typically do a rebase in this situation. Using the `--interactive/-i` flag, I will remove all the other commits except for my own. In this case, I see you have commits `a0b57b4b` and `1a2b1966`.

```
$ git checkout develop
$ git pull
$ git checkout <your_branch>
$ git rebase -i develop
... an editor will pop open in your terminal ...
... remove any line that's not your commits ...
```

**Review by yufengwng — DISMISSED** [review]
